apm-agent-core/src/main/java/co/elastic/apm/agent/impl/stacktrace/StacktraceConfiguration.java

```diff
@@ -91,4 +109,8 @@ public int getStackTraceLimit() {
     public long getSpanFramesMinDurationMs() {
         return spanFramesMinDurationMs.getValue().getMillis();
     }
+
+    public long getSpanStackTraceMinDurationMs() {
+        return spanStackTraceMinDurationMs.getValue().getMillis();
+    }
```

Comment:
Remove `getSpanFramesMinDurationMs` and check both options in `getSpanStackTraceMinDurationMs`. If only `span_frames_min_duration ` is set (check with `isDefault()`),  swap -1 and 0 and return it. Else return `span_stack_trace_min_duration`.

Reply:
fixed